spec/controllers/hq/hq_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+class NyuksController < HqController; end
+
+RSpec.describe HqController, type: :controller do
+  describe 'owns namespace controllers' do
+    Rails.application.eager_load!
+    ApplicationController.descendants.each do |_controller|
+      if _controller.to_s.include? 'Hq::'
+        specify "HqController is the parent of #{_controller.to_s}" do
+          expect(_controller.superclass).to eq HqController
+        end
+      end
+    end
+  end
+
+  controller NyuksController do
+    def foobar_action
+      render nothing: true
+    end
+  end
+
+  describe 'anonymous inherited controller' do
+    specify 'checks for authentication' do
```

Comment:
This is good enough and I'm happy for it to be merged whether changed or not. Being really picky though
`anonymous inherited controller => checks for authentication` suggest that it only checks for authentication for this controller.
I'm wondering whether it should be something like
`all controllers inheriting from hqcontroller are checked for authenticaion`